ambry-store/src/main/java/com.github.ambry.store/CompactAllPolicyFactory.java

```diff
@@ -49,7 +49,7 @@ public CompactionPolicy getCompactionPolicy() {
   CompactAllPolicy(StoreConfig storeConfig, Time time) {
     this.storeConfig = storeConfig;
     this.time = time;
-    this.messageRetentionTimeInMs = storeConfig.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
+    this.messageRetentionTimeInMs = (long) storeConfig.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
```

Comment:
This is being fixed in a different way (using the preferred `TimeUnit` utilities) as part of [this](https://github.com/linkedin/ambry/pull/617/files#diff-fad74bb47f2e54088777e34754ff84d6L52) patch.